core/src/main/java/com/oath/oak/EntrySet.java

```diff
@@ -930,30 +692,30 @@ boolean deleteValueFinish(LookUp lookUp) {
      * therefore the key is written to off-heap and associated with the entry simultaneously.
      * The value of the new entry is set to NULL: <INVALID_VALUE_REFERENCE, INVALID_VERSION></>
      **/
-    LookUp allocateEntry(K key) {
+    void allocateEntry(ThreadContext ctx, K key) {
+        ctx.invalidate();
+
         int ei = nextFreeIndex.getAndIncrement();
         if (ei > entriesCapacity) {
-            return null;
+            return;
         }
         numOfEntries.getAndIncrement();
         int intIdx = entryIdx2intIdx(ei);
         // key and value must be set before returned
         // setting the value reference to <INVALID_VALUE_REFERENCE, INVALID_VERSION>
         // TODO: should we do the following setting? Because it is to set zero on zero...
-        setEntryFieldLong(intIdx, OFFSET.VALUE_REFERENCE, INVALID_VALUE_REFERENCE);
-        setEntryFieldInt(intIdx, OFFSET.VALUE_VERSION, INVALID_VERSION);
+        setEntryFieldLongVolatile(intIdx, OFFSET.VALUE_REFERENCE, ReferenceCodec.INVALID_REFERENCE);
```

Comment:
I do not understand why Volatile... specifically here...?

Reply:
This PR is intentionally avoiding changes in the logic.
`setEntryFieldLong()` was renamed to `setEntryFieldLongVolatile()` because it used volatile access and an additional `setEntryFieldLong()` was added that does not use volatile to write the value.
So this change is merely a rename.

Reply:
This comment is directly related to another comment above on `setEntryFieldLong()`.
I suggest opening an issue regarding the necessity of volatile access here.